components/api-gateway-migrator/cmd/main.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"path/filepath"
+	"strings"
+
+	gatewayv1alpha1 "github.com/kyma-incubator/api-gateway/api/v1alpha1"
+	oldapi "github.com/kyma-project/kyma/components/api-controller/pkg/apis/gateway.kyma-project.io/v1alpha2"
+	"github.com/kyma-project/kyma/components/api-gateway-migrator/pkg/finder"
+	"github.com/kyma-project/kyma/components/api-gateway-migrator/pkg/migrator"
+	rulev1alpha1 "github.com/ory/oathkeeper-maester/api/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	networkingv1alpha3 "knative.dev/pkg/apis/istio/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	_ = clientgoscheme.AddToScheme(scheme)
+	_ = gatewayv1alpha1.AddToScheme(scheme)
+	_ = networkingv1alpha3.AddToScheme(scheme)
+	_ = rulev1alpha1.AddToScheme(scheme)
+	_ = oldapi.AddToScheme(scheme)
+}
+
+func main() {
+	var omitApisWithLabels string
+	flag.StringVar(&omitApisWithLabels, "omit-apis-with-labels", "", "Comma-separated list of keys or key=value pairs defining labels of objects that should be omitted. If only a key is provided, then any value will be matched.")
+	flag.Parse()
+	labels := parseLabels(omitApisWithLabels)
+
+	kubeConfig := initKubeConfig()
+
+	copts := client.Options{
+		Scheme: scheme,
+	}
+
+	k8sClient, err := client.New(kubeConfig, copts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// TODO: pass config from flags
+	defaultConfig := migrator.Config{
+		RetriesCount:        3,
+		DelayBetweenSteps:   2,
+		DelayBetweenRetries: 2,
+	}
+
+	clientWrapper := migrator.NewClient(k8sClient, defaultConfig)
+	f := finder.New(clientWrapper, labels)
+
+	apisToMigrate, err := f.Find()
+
+	switch len(apisToMigrate) {
```

Comment:
This could be swapped for `if len() >= 1; x objects to migrate; else no objects to migrate`

Reply:
But then if you wanted to migrate just one api, the log would read "1 objects to migrate", which is simply outrageous! I suggest something like "number of apis to migrate: x"

Reply:
Suggestion applied